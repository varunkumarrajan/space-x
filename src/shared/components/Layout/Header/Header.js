// React
import React from 'react';
// Prop Types
import PropTypes from 'prop-types'; // ES6
// Styled component
import styled from 'styled-components';

// Styled header wrapper
const HeaderWrapper = styled.header`
  font-size: 24px;
  font-weight: bolder;
  margin: 0 10px;
`;

/**
 * @description html semantic header
 * @param {object} {content}
 */
const Header = ({ content }) => (
  <HeaderWrapper data-testid="headerId">{content.heading}</HeaderWrapper>
);

// Default header
export default Header;

// Default props
Header.defaultProps = {
  content: {
    heading: '',
  },
};
// Prop types
Header.propTypes = {
  content: PropTypes.shape({
    heading: PropTypes.string.isRequired,
  }).isRequired,
};
