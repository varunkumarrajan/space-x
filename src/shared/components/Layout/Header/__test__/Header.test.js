// React
import React from 'react';
// Testing library
import { render, cleanup, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Header from './../Header';

// Mock data
const headerMock = {
    heading: 'testing'
}

afterEach(cleanup);

describe('Header', () => {
  it('Should Render Header with props', () => {
    const { getByTestId } = render(<Header content={headerMock} />);
    const headerId = getByTestId('headerId');
    expect(headerId.className).toBeTruthy();
    const element = screen.getByText(/testing/i);
    expect(element).toBeInTheDocument();
  });

  it('Should Header Snapshot', () => {
    const header = renderer.create(<Header content={headerMock} />).toJSON();
    expect(header).toMatchSnapshot();
  });
});
