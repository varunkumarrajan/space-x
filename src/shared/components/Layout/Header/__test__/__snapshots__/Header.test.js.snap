// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should Header Snapshot 1`] = `
<header
  className="sc-bdfBwQ jIqIGL"
  data-testid="headerId"
>
  testing
</header>
`;
